collector/receiver/otelarrowreceiver/internal/arrow/arrow.go

```diff
@@ -552,235 +730,177 @@ func (r *Receiver) flushSender(serverStream anyStreamServer, pendingCh <-chan ba
 }
 
 func (r *Receiver) srvSendLoop(ctx context.Context, serverStream anyStreamServer, pendingCh <-chan batchResp) error {
-	var err error
 	for {
 		select {
 		case <-ctx.Done():
-			err = r.flushSender(serverStream, pendingCh)
-			// err := multierr.Append(err, ctx.Err())
-			return err
+			return r.flushSender(serverStream, pendingCh)
 		case resp := <-pendingCh:
-			err = r.sendOne(serverStream, resp)
-			if err != nil {
+			if err := r.sendOne(serverStream, resp); err != nil {
 				return err
 			}
 		}
 	}
 }
 
-func (r *Receiver) processAndConsume(ctx context.Context, method string, arrowConsumer arrowRecord.ConsumerAPI, req *arrowpb.BatchArrowRecords, authErr error, response batchResp, sizeHeaderFound bool) error {
-	var err error
-
-	ctx, span := r.tracer.Start(ctx, "otel_arrow_stream_recv")
-	defer span.End()
-
-	r.recvInFlightRequests.Add(ctx, 1)
-	defer func() {
-		r.recvInFlightRequests.Add(ctx, -1)
-		// Set span status if an error is returned.
-		if err != nil {
-			span := trace.SpanFromContext(ctx)
-			span.SetStatus(otelcodes.Error, err.Error())
-		}
-	}()
-
-	// Process records: an error in this code path does
-	// not necessarily break the stream.
-	if authErr != nil {
-		err = authErr
-	} else {
-		err = r.processRecords(ctx, method, arrowConsumer, req, response, sizeHeaderFound)
-	}
-
-	return err
-}
+// consumeBatch applies the batch to the Arrow Consumer, returns a
```

Comment:
Reviewer notes: The function body in the old code was once large and repetitive. The key bugfix here is to split this function into two parts, one that runs sequentially to consume Arrow data, and one that runs concurrently for the pipeline to process data.

There is less repetition now, due to the consolidation into `inFlightData`.